### PR TITLE
seminars: add builder

### PR DESCRIPTION
**Pull request by harunurhan**
INSPIR-3493

### CODE REVIEW

File: inspire_schemas/builders/seminars.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+            latitude (float): latitude of the location.
+            longitude (float): longitude of the location.
+            place_name (str): name of the specific place where this is located.
+            postal_address (str): full postal address in original language.
+            postal_code (str): postal code of the location.
+            state (str): state or province of the location.
+        """
+        address = {}
+
+        if cities:
```
Comment (harunurhan) [inline inspire_schemas/builders/seminars.py:89]: do we have a better way of doing this in place already? @michamos

Comment (michamos) [reply inspire_schemas/builders/seminars.py:89]: I don't think so

File: inspire_schemas/builders/seminars.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    def add_series(self, name, number=None):
+        """Add conference series.
+
+        Args:
+            name (str): name of the conference series.
+            number (int): number of the conference series.
+        """
+        serie_object = self._sourced_dict(name=name)
+        if number:
+            serie_object['number'] = number
```
Comment (michamos) [inline inspire_schemas/builders/seminars.py:210]: I think `_sourced_dict` already filters empty values.

File: inspire_schemas/builders/seminars.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+            title (str): title for the current document.
+            subtitle (str): subtitle for the current document.
+            source (str): source for the given title.
+        """
+        title = self._sourced_dict(
+            source,
+            title=title,
+        )
+        if subtitle is not None:
+            title['subtitle'] = subtitle
```
Comment (michamos) [inline inspire_schemas/builders/seminars.py:227]: same

File: inspire_schemas/builders/seminars.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        """
+        Args:
+            name (str): name of the contact.
+            record (dict): dictionary with ``$ref`` pointing to proper record.
+            If string, then will be converted to proper dict.
+            affiliations (list): list of affiliations objects
+        """
+        self._append_to(
+            'speakers',
+            name=name,
```
Comment (michamos) [inline inspire_schemas/builders/seminars.py:138]: you should either use the signature builder here or manually call `normalize_name`.

File: inspire_schemas/builders/seminars.py [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+
+    def set_end_datetime(self, date=None):
+        """
+        Args:
+            date (str)
+        """
+        if date is not None:
+            self.record['end_datetime'] = date
+
+    def set_start_datetime(self, date=None):
```
[the remark this answers is not in the document]
Comment (harunurhan) [reply inspire_schemas/builders/seminars.py:250]: I didn't want var name to overlap with native `datetime` module.

File: inspire_schemas/builders/seminars.py (hunk 2 above)
Comment (harunurhan) [reply inspire_schemas/builders/seminars.py:210]: I copy pasted this from `conferences` actually, but I can't see how it filters:

```python
    def _sourced_dict(self, source=None, **kwargs):
        if source:
            kwargs['source'] = source
        elif self.source:
            kwargs['source'] = self.source
        return kwargs
```

Comment (michamos) [reply inspire_schemas/builders/seminars.py:211]: sorry, I was confused with `_append_to`. Maybe it makes sense doing the check there?

File: inspire_schemas/builders/seminars.py [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+        """
+        Args:
+            name (str): name of the contact.
+            record (dict): dictionary with ``$ref`` pointing to proper record.
+            If string, then will be converted to proper dict.
+            affiliations (list): list of affiliations objects
+        """
+        self._append_to(
+            'speakers',
+            name=normalize_name(name),
```
Comment (michamos) [inline inspire_schemas/builders/seminars.py:139]: can you add a test for name normalization?

Comment (harunurhan) [reply inspire_schemas/builders/seminars.py:139]: it's in the unit tests.

File: inspire_schemas/builders/seminars.py (hunk 2 above)
Comment (harunurhan) [reply inspire_schemas/builders/seminars.py:211]: personally I don't like implicit logic in the functions, but sure I can put it there

**Pull request merged**